src/containers/MyNdla/Learningpath/components/FolderResourcePicker.tsx

```diff
@@ -0,0 +1,246 @@
+  },
+});
+
+const TextWrapper = styled("div", {
+  base: {
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "flex-start",
+    gap: "4xsmall",
+    flex: "1",
+  },
+});
+
+const StyledListItemRoot = styled(ListItemRoot, {
+  base: {
+    minHeight: "unset",
+    textAlign: "start",
+  },
+});
+
+const StyledComboboxContent = styled(ComboboxContentStandalone, {
+  base: {
+    maxHeight: "surface.medium",
+  },
+});
+
+const ContentWrapper = styled("div", {
+  base: {
+    boxShadow: "large",
+    padding: "small",
+    backgroundColor: "background.default",
+    borderRadius: "xsmall",
+  },
+});
+
+const StyledComboboxItem = styled(ComboboxItem, {
+  base: {
+    flexWrap: "wrap",
+  },
+});
+
+const LEGAL_RESOURCE_TYPES = ["article", "multidisciplinary", "topic"];
+
+type GQLFolderResourceWithCrumb = GQLFolderResource & { breadcrumb: GQLBreadcrumb[] };
+
+const flattenFolderResources = (folders: GQLFolder[]): GQLFolderResourceWithCrumb[] => {
+  const results: GQLFolderResourceWithCrumb[] = folders
+    .flatMap((folder) =>
+      folder.resources
+        .filter((resource) => LEGAL_RESOURCE_TYPES.includes(resource.resourceType))
+        .map<GQLFolderResourceWithCrumb>((resource) => ({ ...resource, breadcrumb: folder.breadcrumbs })),
+    )
+    .flat();
+
+  folders.forEach((folder) => results.push(...flattenFolderResources(folder.subfolders)));
+
+  return results;
+};
+
+interface ComboboxProps {
+  onResourceSelect: (resource: FolderResource) => void;
+}
+
+export const FolderResourcePicker = ({ onResourceSelect }: ComboboxProps) => {
+  const [open, setOpen] = useState<boolean>(false);
+  const [inputValue, setInputValue] = useState<string>("");
+  const [filteredResources, setFilteredResources] = useState<GQLFolderResourceWithCrumb[] | undefined>(undefined);
+  const [highlightedValue, setHighligtedValue] = useState<string | null>(null);
+
+  const { folders } = useFolders();
+  const resources = useMemo(() => flattenFolderResources(folders), [folders]);
+
+  const { data, loading } = useFolderResourceMetaSearch(
+    resources.map((r) => ({
+      id: r.resourceId,
+      path: r.path,
+      resourceType: r.resourceType,
+    })),
+  );
+
+  const translations = useComboboxTranslations();
+
+  const keyedData = useMemo(() => keyBy(data ?? [], (resource) => `${resource.type}-${resource.id}`), [data]);
+
+  const collection = useMemo(
+    () =>
+      createListCollection({
+        items: filteredResources ?? [],
+        itemToValue: (item) => item?.id,
+        itemToString: (item) => keyedData[`${item?.resourceType}-${item?.resourceId}`]?.title ?? item?.id,
+      }),
+    [filteredResources, keyedData],
+  );
+
+  const handleChange = (e: ComboboxInputValueChangeDetails) => {
+    const filtered = resources.filter((item) =>
+      keyedData[`${item.resourceType}-${item.resourceId}`]?.title.toLowerCase().includes(e.inputValue.toLowerCase()),
+    );
+    setFilteredResources(filtered);
```

Comment:
Du trenger vel ikke å sette filteredResources her?

Reply:
Det er sant! gikk litt fort, jeg tenkte dette var logikk for å sette valgt element 😅